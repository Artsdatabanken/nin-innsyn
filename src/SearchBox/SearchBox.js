import React, { Component } from 'react'
import { TextField } from 'material-ui'
import PropTypes from 'prop-types'

export default class SearchBox extends Component {
  onKeyDown = e => {
    if (e.keyCode === 27) {
      this.props.onQueryChange(e, '')
      this.textField.blur()
      e.stopPropagation()
    }
  }

  handleFocus = e => {
    this.props.onQueryChange(e, this.props.tittel)
  }

  render() {
    const { query, tittel } = this.props
    return (
      <TextField
        ref={tf => (this.textField = tf)}
        onKeyDown={this.onKeyDown}
        value={query}
        hintText={tittel ? tittel : 'Ã˜kologisk grunnkart'}
        onFocus={this.handleFocus}
        onChange={this.props.onQueryChange}
        fullWidth={true}
        underlineShow={false}
        style={{ lineHeight: '24px' }}
        inputStyle={{ color: 'black', fontSize: '15px' }}
        hintStyle={{ color: 'hsl(0, 0%, 75%)', fontSize: '15px' }}
      />
    )
  }
}

SearchBox.propTypes = {
  onSearchResults: PropTypes.func,
}
