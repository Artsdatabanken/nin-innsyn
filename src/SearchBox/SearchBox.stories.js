import React from 'react'
import { storiesOf } from '@storybook/react'
import { muiTheme } from 'storybook-addon-material-ui'
import SearchBox from './SearchBox'

storiesOf('SearchBox', module)
  .addDecorator(muiTheme())
  .add('default', () => <SearchBox />)
