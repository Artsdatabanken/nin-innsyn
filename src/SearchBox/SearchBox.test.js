import React from 'react'
import { shallow } from 'enzyme'
import SearchBox from './SearchBox'

test('searchBox', () => {
  const onQueryChange = jest.fn()
  const wrapper = shallow(<SearchBox onQueryChange={onQueryChange} />)
  expect(wrapper).toMatchSnapshot()

  const textField = wrapper.find('TextField')
  wrapper.instance().textField = { blur: () => {} }
  const event = { keyCode: 27, stopPropagation: () => {} }
  textField.simulate('keyDown', event)
  expect(onQueryChange).toBeCalledWith(event, '')

  expect(wrapper).toMatchSnapshot()
})
