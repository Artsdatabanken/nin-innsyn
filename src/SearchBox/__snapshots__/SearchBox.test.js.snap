// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchBox 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    onQueryChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "disabled": false,
      "floatingLabelFixed": false,
      "fullWidth": true,
      "hintStyle": Object {
        "color": "hsl(0, 0%, 75%)",
        "fontSize": "15px",
      },
      "hintText": "Natur i Norge",
      "id": "1",
      "inputStyle": Object {
        "color": "black",
        "fontSize": "15px",
      },
      "multiLine": false,
      "onChange": [Function],
      "onFocus": [Function],
      "onKeyDown": [Function],
      "rows": 1,
      "style": Object {
        "lineHeight": "24px",
      },
      "type": "text",
      "underlineShow": false,
      "value": "",
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "disabled": false,
        "floatingLabelFixed": false,
        "fullWidth": true,
        "hintStyle": Object {
          "color": "hsl(0, 0%, 75%)",
          "fontSize": "15px",
        },
        "hintText": "Natur i Norge",
        "id": "1",
        "inputStyle": Object {
          "color": "black",
          "fontSize": "15px",
        },
        "multiLine": false,
        "onChange": [Function],
        "onFocus": [Function],
        "onKeyDown": [Function],
        "rows": 1,
        "style": Object {
          "lineHeight": "24px",
        },
        "type": "text",
        "underlineShow": false,
        "value": "",
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`searchBox 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    onQueryChange={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "disabled": false,
      "floatingLabelFixed": false,
      "fullWidth": true,
      "hintStyle": Object {
        "color": "hsl(0, 0%, 75%)",
        "fontSize": "15px",
      },
      "hintText": "Natur i Norge",
      "id": "1",
      "inputStyle": Object {
        "color": "black",
        "fontSize": "15px",
      },
      "multiLine": false,
      "onChange": [Function],
      "onFocus": [Function],
      "onKeyDown": [Function],
      "rows": 1,
      "style": Object {
        "lineHeight": "24px",
      },
      "type": "text",
      "underlineShow": false,
      "value": "",
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "disabled": false,
        "floatingLabelFixed": false,
        "fullWidth": true,
        "hintStyle": Object {
          "color": "hsl(0, 0%, 75%)",
          "fontSize": "15px",
        },
        "hintText": "Natur i Norge",
        "id": "1",
        "inputStyle": Object {
          "color": "black",
          "fontSize": "15px",
        },
        "multiLine": false,
        "onChange": [Function],
        "onFocus": [Function],
        "onKeyDown": [Function],
        "rows": 1,
        "style": Object {
          "lineHeight": "24px",
        },
        "type": "text",
        "underlineShow": false,
        "value": "",
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
